src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -405,18 +406,22 @@ protected void initOutput(BatchSinkContext context, BigQuery bigQuery, String ou
     Configuration configuration = new Configuration(baseConfiguration);
 
     // Build GCS storage path for this bucket output.
+    DatasetId datasetId = DatasetId.of(datasetProject, dataset);
     String temporaryGcsPath = BigQuerySinkUtils.getTemporaryGcsPath(bucket, runUUID.toString(), tableName);
-    BigQuerySinkUtils.configureOutput(configuration,
-      DatasetId.of(datasetProject, dataset),
-      tableName,
-      temporaryGcsPath,
-      fields);
+    BigQuerySinkUtils.configureOutput(configuration, datasetId, tableName, temporaryGcsPath, fields);
     // Both emitLineage and setOutputFormat internally try to create an external dataset if it does not already exist.
     // We call emitLineage before since it creates the dataset with schema which is used.
     List<String> fieldNames = fields.stream()
       .map(BigQueryTableFieldSchema::getName)
       .collect(Collectors.toList());
-    BigQuerySinkUtils.recordLineage(context, outputName, tableSchema, fieldNames);
+    String fqn = BigQueryUtil.getFQN(datasetProject, dataset, config.getTable());
+    String location = bigQuery.getDataset(datasetId).getLocation();
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+      ? ReferenceNames.normalizeFqn(fqn)
+      : config.getReferenceName();
+    io.cdap.plugin.common.Asset lineageAsset = io.cdap.plugin.common.Asset.builder(referenceName)
```

Comment:
Let's import the package at the beginning of the file

Reply:
There's already an `Asset` package imported [here](https://github.com/data-integrations/google-cloud/blob/f5f29fe038f9acefebc924ddd4493d593df83788/src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java#L30)